### ISSUE TITLE
newfloat.sty is very new

### ISSUE THREAD

**Issue by birkenfeld** [sphinx-doc/sphinx#2378]
I noticed that newfloat.sty (which is used in newer 1.3.x releases) is not present in the texlive packages of Ubuntu 12.04. This is unfortunate.

If we really need this package, can we include it?

**Comment by tk0miya** [comment]
Of course, Yes. I'll bundle it later.
It comes from #2147. and I think it is important to insert page breaks into literal_block.

BTW, newfloat.sty is included ubuntu-trusty (14.04). When will we remove bundled styles from our package? At the EOL of 12.04?

**Comment by birkenfeld** [comment]
I don't want to give any concrete date, but 4 years is not very long in terms of TeX progress :)
Something between 5-6 years seems acceptable.

**Comment by tk0miya** [comment]
Okay.
Now I added `newfloat.sty` into `stable` and `1.4-release` branch.

**Comment by rddaz2013** [comment]
It would be nice if all new dependencies 
all *.sty would be in the sphinx package?

Old modified sphinx installations have very much to struggle with the new dependencies, especially when no message is created in the latex build where the change comes from.

oh nice ...\\SetupFloatEnviroment{literal-block} -> undefined control sequence.

**Comment by jfbu** [comment]
@rddaz2013 sorry about your troubles, but it is not caused by a new dependency but by a dropped one.

You find in [CHANGES](http://www.sphinx-doc.org/en/stable/changes.html) mention of the dropped dependencies. There is this entry for 1.5 release:

> * LaTeX package newfloat (which was shipped with Sphinx since 1.3.4) is no
>  longer a dependency of sphinx.sty (ref #2660)

There is actually an error in this CHANGES entry because although `newfloat` became a dependency of Sphinx at 1.3.4 it started being shipped with it only at Sphinx 1.4.

It was dropped as dependency from Sphinx at 1.5, hence it is not shipped with Sphinx anymore. I understand that not shipping external files anymore can create issues for people who used the resources provided by those files, but in the case at hand, generally speaking it is recommended to have a reasonably complete TeX installation to run Sphinx. We struggle to not require more than what comes with a standard Debian/TeXLive2009 install, but will raise requirements to Debian/TeXLive2013 at 1.6 release.

Your set-up would not create errors on a reasonably complete TeX install. `newfloat.sty` is part of the `caption` package (http://www.ctan.org/pkg/caption) and here is how it is described in TeXLive 2016:

```
$ tlmgr info caption
package:     caption
category:    Package
shortdesc:   Customising captions in floating environments
longdesc:    The caption package provides many ways to customise the captions in floating environments like figure and table, and cooperates with many other packages. Facilities include rotating captions, sideways captions, continued captions (for tables or figures that come in several parts). A list of compatibility notes, for other packages, is provided in the documentation. The package also provides the "caption outside float" facility, in the same way that simpler packages like capt-of do. The package supersedes caption2.
installed:   Yes
revision:    41409
sizes:       src: 1193k, doc: 1825k, run: 209k
relocatable: No
cat-date:    2016-05-22 21:27:06 +0200
cat-license: lppl1.3
cat-topics:  caption float
cat-related: subcaption bicaption
collection:  collection-latexrecommended
```

from which we see it is part of ``collection-latexrecommended`` which obviously should be part of any reasonable TeX install. Unfortunately it is a notorious fact that for many years Linux Distros have shipped with antiquated or very lacunary TeX installs.
